### PR TITLE
Implemented adding and deleting techs

### LINKED ISSUE #6: Add a form to add and remove techs

### PR DESCRIPTION
**Pull request by debskipawel**
- Reformatted the main view into 2 components,
- Added fuctionality to delete techs to the main view
- Implemented dialog to add new techs

### CODE REVIEW

File: src/components/AddTechDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  })
+);
+
+export interface AddTechDialogProps {
+  setTechs: (value: React.SetStateAction<User[]>) => void;
+}
+
+const AddTechDialog = (props: AddTechDialogProps) => {
+  const classes = useStyles();
+  const [open, setOpen] = React.useState(false);
```
Comment (chlebnymaciej) [inline src/components/AddTechDialog.tsx]: please add type <boolean>

File: src/components/AddTechDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        {
+          enqueueSnackbar("Could not add tech", { variant: "error" });
+        }
+        else if(response.data)
+        {
+          props.setTechs(prev => 
+            response.data ? 
+            prev = [...prev, response.data]
+            :          
+            prev
```
Comment (chlebnymaciej) [inline src/components/AddTechDialog.tsx]: I think it looks better in one line

**Review by chlebnymaciej — COMMENTED** [review]
Please do add these types in <>

File: src/components/AddTechDialog.tsx (hunk 1 above)
Comment (ArturChmura) [inline src/components/AddTechDialog.tsx]:
```suggestion
  const [open, setOpen] = React.useState<boolean>(false);
```

File: src/components/AddTechDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  const handleClickClose = () => {
+    setOpen(false);
+
+    setUsername('');
+    setPassword('');
+    setConfirmPassword('');
+    setSubmitted(false);
+  }
+
+  const { enqueueSnackbar } = useSnackbar();
```
Comment (ArturChmura) [inline src/components/AddTechDialog.tsx]: move it to the very top of the component.

File: src/components/AddTechDialog.tsx (hunk 2 above)
Comment (ArturChmura) [reply src/components/AddTechDialog.tsx]: it looks even beter without `?` becouse `response.data` is true (you are checking it in else if above)

File: src/components/AddTechDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+  const handleUsernameChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setUsername(event.target.value as string);
```
Comment (ArturChmura) [inline src/components/AddTechDialog.tsx]:
```suggestion
  const handleUsernameChange = (event: React.ChangeEvent<{ value: string}>) => {
    setUsername(event.target.value);
```

File: src/components/AddTechDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  };
+
+  const errorUsername = () => {
+    return username === '';
+  }
+
+  const handlePasswordChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setPassword(event.target.value as string);
```
Comment (ArturChmura) [inline src/components/AddTechDialog.tsx]:
```suggestion
  const handlePasswordChange = (event: React.ChangeEvent<{ value: string}>) => {
    setPassword(event.target.value);
```

File: src/components/AddTechDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+  };
+
+  const errorPassword = () => {
+    console.log("Password: " + password)
+    return password === '';
+  }
+
+  const handleConfirmPasswordChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setConfirmPassword(event.target.value as string);
```
Comment (ArturChmura) [inline src/components/AddTechDialog.tsx]:
```suggestion
  const handleConfirmPasswordChange = (event: React.ChangeEvent<{ value: string}>) => {
    setConfirmPassword(event.target.value);
```

File: src/components/AddTechDialog.tsx (hunk 2 above)
Comment (ArturChmura) [reply src/components/AddTechDialog.tsx]: also never assign values to prev

Comment (ArturChmura) [reply src/components/AddTechDialog.tsx]:
```suggestion
        const data = response.data;
        props.setTechs(prev => [...prev, data])
```

**Review by ArturChmura — CHANGES_REQUESTED** [review]

File: src/components/AddTechDialog.tsx (hunk 2 above)
Comment (debskipawel) [reply src/components/AddTechDialog.tsx]: Fixed it just like you suggested

File: src/components/AddTechDialog.tsx (hunk 1 above)
Comment (debskipawel) [reply src/components/AddTechDialog.tsx]: Done!

Comment (debskipawel) [reply src/components/AddTechDialog.tsx]: Done!

File: src/components/AddTechDialog.tsx (hunk 3 above)
Comment (debskipawel) [reply src/components/AddTechDialog.tsx]: Done!

File: src/components/AddTechDialog.tsx (hunk 4 above)
Comment (debskipawel) [reply src/components/AddTechDialog.tsx]: Done!

File: src/components/AddTechDialog.tsx (hunk 6 above)
Comment (debskipawel) [reply src/components/AddTechDialog.tsx]: Done!

File: src/components/AddTechDialog.tsx (hunk 5 above)
Comment (debskipawel) [reply src/components/AddTechDialog.tsx]: Done!

**Comment by debskipawel** [comment]
Applied all the requested changes, please take a look @ArturChmura 👀

**Comment by debskipawel** [comment]
> Applied all the requested changes, please take a look @ArturChmura 👀

@chlebnymaciej you can also take a look 😅

**Review by ArturChmura — APPROVED** [review]
Well done

**Pull request merged**